### DISCUSSION TITLE
XYChart: Handling gaps in data

### DISCUSSION THREAD

**Discussion by RyKilleen** [airbnb/visx#1727] [Q&A] [answered]
Hello and thanks for an excellent library!

I'm curious if there's a definitive way to handle gaps in data with `visx`, specifically the `XYChart`? 

For example, consider a time series that may have periods of time without data. The following example shows how it works today: 
![image](https://github.com/airbnb/visx/assets/4615840/1e9f2b22-2723-4363-bdad-76a50e62ee05)

The path continues based on the curve to the next valid data point. Is it possible to have a gap in the line instead?

**Comment by williaster** [comment, marked as answer]
Hey @RyKilleen thanks for checking out `visx`. To create gaps in your `LineSeries` all you'll need to do is fill in the missing `x` values in your dataset and set the corresponding `y` values to `null`. 

For example
```ts
[
  { x: '2023-04-30', y: 10 },
  { x: '2023-05-01', y: 10 },
  { x: '2023-05-02', y: null }, // add something like this
  { x: '2023-05-03', y: null }, // add something like this
  { x: '2023-05-04', y: 10 },
]
```

That should accomplish something like this ([link](https://codesandbox.io/s/xychart-data-gaps-dwmfjh?file=/Example.tsx)). I would note that you must set [`stroke-linecap=round`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linecap#usage_notes) in order to see **single points** surrounded by missing data (this is the default in `XYChart`, but mentioning in case anyone overrides this)

<img width="1326" alt="image" src="https://github.com/airbnb/visx/assets/4496521/b3fb440b-d8a9-420b-9ae6-2d9ff3c85039">

**Reply by RyKilleen** [reply]
Thanks so much, appreciate it!
